chia/wallet/wallet_state_manager.py

```diff
@@ -874,16 +886,12 @@ async def handle_cat(
             return None
         else:
             our_inner_puzzle: Program = self.main_wallet.puzzle_for_pk(derivation_record.pubkey)
-            asset_id: bytes32 = bytes32(bytes(tail_hash)[1:])
+            asset_id: bytes32 = parent_data.tail_program_hash
             cat_puzzle = construct_cat_puzzle(CAT_MOD, asset_id, our_inner_puzzle, CAT_MOD_HASH)
             if cat_puzzle.get_tree_hash() != coin_state.coin.puzzle_hash:
                 return None
-            if bytes(tail_hash).hex()[2:] in self.default_cats or self.config.get(
-                "automatically_add_unknown_cats", False
-            ):
-                cat_wallet = await CATWallet.get_or_create_wallet_for_cat(
-                    self, self.main_wallet, bytes(tail_hash).hex()[2:]
-                )
+            if asset_id.hex() in self.default_cats or self.config.get("automatically_add_unknown_cats", False):
+                cat_wallet = await CATWallet.get_or_create_wallet_for_cat(self, self.main_wallet, asset_id.hex())
```

Comment:
It doesn't change anything. Since bytes(tail_hash).hex() will return a0+actual hex, so we want to truncate the first 2 chars. But parent_data.tail_program_hash already truncated "a0" by using .atom, so we don't need [2:] any more.